### PR TITLE
Mapper: index field implemented

### PR DESCRIPTION
**Pull request by pvanek**
@davidnich @tethal please review. There are still some open questions:
- should index accept integers only or is it fine to use Qore's dynamic type features? Example: "indextest3" : "a" results in "a0", "a1", ... which is sort of strange, but it can be useful in some cases.
- @gamato mapper.qtest, testMapperOptionInput does not contain any assertion at all

### CODE REVIEW

**Comment by pvanek** [comment]
@pavelkveton review too as it can be used in your MIPs...

File: qlib/Mapper.qm [hunk 1]
```diff
@@ -237,6 +238,7 @@ m.mapData(input2);           # date_begin is still the same as from beginning of
       - @ref Mapper::Mapper::getRuntime()
       - @ref Mapper::Mapper::replaceRuntime()
       - @ref Mapper::Mapper::setRuntime()
+    - implemented \c "index" fiels tag for current row index
```
Comment (davidnich) [inline qlib/Mapper.qm]: please fix the typo here

**Comment by davidnich** [comment]
three minor issues:
- please fix the typo @ Mapper.qm:241
- please add a release note in the Mapper module itself under "@subsection mapperv1_1 Mapper v1.1"
- please add a release note in doxygen/lang/release_notes_900.dox.tmpl under "Mapper module updates"

otherwise looks good, I'll assign to Pavel now for review, @pavelkveton when you are done, please assign ondra for merging

**Comment by pvanek** [comment]
ok, typo fixed, release notes added (BTW that typo was in "@subsection mapperv1_1 Mapper v1.1" already

**Comment by tethal** [comment]
:+1:

**Comment by pavelkveton** [comment]
Looks nice, thanks, Petr
Reassigning to Ondrej to merge

**Comment by gamato** [comment]
Hi. A bit late with commenting, but still ...
- "should index accept integers only or is it fine to use Qore's dynamic type features" -- I'd go with the latter, index being not a pure integer may be useful.
- "mapper.qtest, testMapperOptionInput does not contain any assertion at all" -- true, but the test would trigger old bug in Mapper that I fixed.
- mapper.qtest, I specifically changed opts / m_opts so that its empty when setUp() is not called. Now you moved stuff from setUp() to private{}. I believe that tests should be independent and so if e.g. runtime feature gets broken other tests should pass. So private{} / setUP() should contain just stuff that's necessary to and used by all tests.

Otherwise it looks good. Thanks.

**Pull request merged**